docs/rules/max-lines-per-function.md

```diff
@@ -0,0 +1,130 @@
+# enforce a maximum function length (max-lines-per-function)
+
+Some people consider large functions a code smell. Large functions tend to do a lot of things and can make it hard following what's going on. While there is not an objective maximum number of lines considered acceptable in a function, most people would agree it should not be above 200. Google's JS style guide (by proxy of their C++ style guide), recommends 40 lines or fewer per function. The Linux Kernel's style guide recommends 48 lines or fewer.
+
+## Rule Details
+
+This rule enforces a maximum number of lines per function, in order to aid in maintainability and reduce complexity. This rule does not count any top-level function code.
```

Comment:
It might be worth referencing [`max-lines`](https://eslint.org/docs/rules/max-lines) here specifically for the case of top-level code. Also, "top-level function code" is ambiguous here: does it include either/both of these?

```js
// 1.
function foo() {
    // ...
}

// 2.
;(function () {
    // ...
})()
```

Reply:
Any function declaration will by included, so yes, both your examples will be counted. "top level function" isn't the right name